gems/aws-sdk-s3/lib/aws-sdk-s3/multipart_file_uploader.rb

```diff
@@ -113,8 +112,12 @@ def has_checksum_key?(keys)
         keys.any? { |key| checksum_key?(key) }
       end
 
+      def checksum_not_required?(options)
+        @client.config.request_checksum_calculation == 'when_required' && !options[:checksum_algorithm]
```

Comment:
In general consider old s3 new core case.

Reply:
Talked offline